### PR TITLE
Adds a provider lookup for sitemaps

### PR DESCRIPTION
**Pull request by bigbasec**
This adds a sitemap provider lookup to the proxy service.   

It's one part of a 2 part fix for issues : 
https://github.com/openhab/openhab-webui/issues/763
https://github.com/openhab/openhab-core/issues/2154
https://github.com/openhab/openhab-webui/issues/745

### CODE REVIEW

File: bundles/org.openhab.core.ui/src/main/java/org/openhab/core/ui/internal/proxy/ProxyServletService.java [hunk 1]
```diff
@@ -86,8 +92,19 @@
     public static final String ATTR_URI = ProxyServletService.class.getName() + ".URI";
     public static final String ATTR_SERVLET_EXCEPTION = ProxyServletService.class.getName() + ".ProxyServletException";
 
+    private static final String SITEMAP_SUFFIX = ".sitemap";
+
     private final Logger logger = LoggerFactory.getLogger(ProxyServletService.class);
 
+    private final List<SitemapProvider> sitemapProviders = new ArrayList<>();
+
+    public interface SitemapSubscriptionCallback {
```
Comment (kaikreuzer) [inline bundles/org.openhab.core.ui/src/main/java/org/openhab/core/ui/internal/proxy/ProxyServletService.java:101]: What is this interface good for?

File: bundles/org.openhab.core.ui/src/main/java/org/openhab/core/ui/internal/proxy/ProxyServletService.java [hunk 2]
```diff
@@ -96,6 +113,28 @@
...
     protected ModelRepository modelRepository;
 
+    @Reference(cardinality = ReferenceCardinality.MULTIPLE, policy = ReferencePolicy.DYNAMIC)
+    protected void addSitemapProvider(SitemapProvider provider) {
+        sitemapProviders.add(provider);
+        provider.addModelChangeListener(this);
+    }
+
+    @Override
+    public void modelChanged(String modelName, EventType type) {
```
Comment (kaikreuzer) [inline bundles/org.openhab.core.ui/src/main/java/org/openhab/core/ui/internal/proxy/ProxyServletService.java:123]: Why does the class implement `ModelRepositoryChangeListener`? This method does not seem to do anything besides doing a debug log statement.

File: bundles/org.openhab.core.ui/src/main/java/org/openhab/core/ui/internal/proxy/ProxyServletService.java [hunk 3]
```diff
@@ -96,6 +113,28 @@
     protected ItemUIRegistry itemUIRegistry;
     protected ModelRepository modelRepository;
```
Comment (kaikreuzer) [inline bundles/org.openhab.core.ui/src/main/java/org/openhab/core/ui/internal/proxy/ProxyServletService.java:114]: The `modelRepository` does not seem to be used anymore, so I'd say it should be removed.

**Review by kaikreuzer — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.core.ui/src/main/java/org/openhab/core/ui/internal/proxy/ProxyServletService.java (hunk 3 above)
Comment (bigbasec) [reply bundles/org.openhab.core.ui/src/main/java/org/openhab/core/ui/internal/proxy/ProxyServletService.java:114]: Leftover from testing.   Removed.

File: bundles/org.openhab.core.ui/src/main/java/org/openhab/core/ui/internal/proxy/ProxyServletService.java (hunk 2 above)
Comment (bigbasec) [reply bundles/org.openhab.core.ui/src/main/java/org/openhab/core/ui/internal/proxy/ProxyServletService.java:123]: Leftover from testing.   Removed.

File: bundles/org.openhab.core.ui/src/main/java/org/openhab/core/ui/internal/proxy/ProxyServletService.java (hunk 1 above)
Comment (bigbasec) [reply bundles/org.openhab.core.ui/src/main/java/org/openhab/core/ui/internal/proxy/ProxyServletService.java:101]: Leftover from testing.   Removed.

**Review by bigbasec — COMMENTED** [review]
All suggestions have been implemented.   Thank you!

File: bundles/org.openhab.core.ui/src/main/java/org/openhab/core/ui/internal/proxy/ProxyServletService.java [hunk 4]
```diff
@@ -85,15 +92,28 @@
     public static final String ATTR_URI = ProxyServletService.class.getName() + ".URI";
     public static final String ATTR_SERVLET_EXCEPTION = ProxyServletService.class.getName() + ".ProxyServletException";
 
+    private static final String SITEMAP_SUFFIX = ".sitemap";
```
Comment (kaikreuzer) [inline bundles/org.openhab.core.ui/src/main/java/org/openhab/core/ui/internal/proxy/ProxyServletService.java:95]: Another left-over?

**Comment by kaikreuzer** [comment]
Thanks for your updates.
Please run 
```
mvn spotless:apply
```
as adviced by the [failing Jenkins build](https://ci.openhab.org/job/PR-openHAB-Core/3102/console).

File: bundles/org.openhab.core.ui/src/main/java/org/openhab/core/ui/internal/proxy/ProxyServletService.java (hunk 4 above)
Comment (bigbasec) [reply bundles/org.openhab.core.ui/src/main/java/org/openhab/core/ui/internal/proxy/ProxyServletService.java:95]: Yes sir, thank you!

**Comment by bigbasec** [comment]
> Thanks for your updates.
> Please run
> 
> ```
> mvn spotless:apply
> ```
> 
> as adviced by the [failing Jenkins build](https://ci.openhab.org/job/PR-openHAB-Core/3102/console

> Thanks for your updates.
> Please run
> 
> ```
> mvn spotless:apply
> ```
> 
> as adviced by the [failing Jenkins build](https://ci.openhab.org/job/PR-openHAB-Core/3102/console).

I actually didn't do that because on a clean checkout the first time you run that it'll change a number of files that aren't really being changed.   Something with the way Eclipse does what it does I would suspect.   If I do that now and push those changes this modification goes from 1 file to 11, which is not the desired commit.   

I'm switching to VS Code, hopefully have better results.

**Comment by kaikreuzer** [comment]
>  it'll change a number of files that aren't really being changed

Then only commit the change that concerns the class you are touching here and all should be fine.

**Review by bigbasec — COMMENTED** [review]
Removed as per review.

**Comment by bigbasec** [comment]
Should be all set now, I hope.   Really need to learn more about Git, I'm terrible with this.

**Comment by kaikreuzer** [comment]
Jenkins is still failing. Seems you didn't yet push the commit with the spotless fixes.

**Comment by bigbasec** [comment]
I only committed the one file, so I don't know.   Ran the spotless:apply committed and that's all.   I'll try again I guess.

**Comment by bigbasec** [comment]
Just ran it again and there would be no changes to the file in question, so I don't know what to do here.   Sorry!

**Comment by kaikreuzer** [comment]
There must be changes, otherwise Jenkins wouldn't fail. Does the Maven build succeed locally for you? Are you sure that you have pushed all your changes?

**Comment by bigbasec** [comment]
Nope, completely broken by the updates above to satisfy your recommendations.   Give me a minute I'll fix all that.

**Comment by bigbasec** [comment]
5th time's a charm.  Apologies, I'll get this sooner or later.

File: bundles/org.openhab.core.ui/src/main/java/org/openhab/core/ui/internal/proxy/ProxyServletService.java (hunk 3 above)
Comment (bigbasec) [reply bundles/org.openhab.core.ui/src/main/java/org/openhab/core/ui/internal/proxy/ProxyServletService.java:102]: And I was wrong, it is used.

**Comment by bigbasec** [comment]
Sorry, I tried.  Again, I give up.

**Pull request closed without merging**